src/sagas/repository.ts

```diff
@@ -787,16 +787,19 @@ export function* watchGetDatasetByIdSuccess(): any {
 
 export function* previewData({ payload }: any): any {
   const queryState = yield select(getQuery);
-  const offset = queryState.page * queryState.rowsPerPage;
-  const limit = queryState.rowsPerPage;
-  const sort = queryState.orderProperty === undefined ? '' : `&sort=${queryState.orderProperty}`;
-  const sortDirection =
-    queryState.orderDirection === undefined ? '' : `&direction=${queryState.orderDirection}`;
-  const filter =
-    queryState.filterStatement === undefined ? '' : `&filter=${queryState.filterStatement}`;
-  const query = `/api/repository/v1/${payload.resourceType}s/${payload.resourceId}/data/${payload.table}?offset=${offset}&limit=${limit}${sort}${sortDirection}${filter}`;
+  const queryDataRequest = {
+    offset: queryState.page * queryState.rowsPerPage,
+    limit: queryState.rowsPerPage,
+    sort: _.isEmpty(queryState.orderProperty) ? DbColumns.ROW_ID : `${queryState.orderProperty}`,
+    direction:
+      queryState.orderDirection === undefined
+        ? SqlSortDirection.Asc
+        : `${queryState.orderDirection}`,
+    filter: queryState.filterStatement === undefined ? '' : `${queryState.filterStatement}`,
```

Comment:
I think you can use `_.isEmpty` to check whether `queryState.orderDirection` and `queryState.filterStatement` are undefined as well

Reply:
Good point!